src/order_tree.cpp

```diff
@@ -0,0 +1,204 @@
+#pragma once
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\order_tree.h"
+
+#include <cmath>
+#include <memory>
+using std::shared_ptr;
+
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\order.h"
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\transactions.h" 
+#include "C:\Users\Hamza Rashid\Desktop\C++\Trading Engine\include\TradingEngine\limit_order.h"
+
+/* Return length of Order's within the Tree */
+int OrderTree::get_length() const {
+	return order_map_.size();
+}
+
+/* Get min price within Ordertree */
+float OrderTree::get_min() {
+    return min_price_;
+}
+
+/* Set min price within OrderTree */
+void OrderTree::set_min(const float& new_min) {
+    min_price_ = new_min;
+}
+
+/* Get max price within Ordertree */
+float OrderTree::get_max() {
+    return max_price_;
+}
+
+/* Set min price within OrderTree */
+void OrderTree::set_max(const float& new_max) {
+    max_price_ = new_max;
+}
+
+/* Return Tree type */
+const OrderType& OrderTree::get_type() const {
+    return type_;
+}
+
+/* Add Order into Tree */
+void OrderTree::add_price_order(const std::shared_ptr<Order>& incoming_order) {
+    
+    /* Check and create the respective LimitOrder */
+    if (!price_map_.count(incoming_order->get_price())) {
```

Comment:
instead of doing this, the standard is to use the method `find` and check that it doesn't equal the price_map.end()